usage/src/main/resources/org/killbill/billing/usage/dao/RolledUpUsageSqlDao.sql.stg

```diff
@@ -17,10 +18,20 @@ tableValues() ::= <<
 , :unitType
 , :recordDate
 , :amount
+, :trackingId
 , :userName
 , :createdDate
 >>
 
+getRecordsCountByTrackingId() ::= <<
```

Comment:
tl;dr: it's probably fine, but we will have to performance test it (and run `EXPLAIN` on it) with a very large table.

Just for the record: because we will insert only one batch at most for a given `subscription_id` / `tracking_id`, this will scan at most a few hundred leaves in the index (unless the DB has some statistics on it). The performance should probably be the same as:

```
select
  1
where subscription_id = :subscriptionId
and tracking_id = :trackingId
<AND_CHECK_TENANT()>
limit 1
;
```

which is the approach I was thinking of intuitively originally (since we don't really need the count).